filebeat/beater/filebeat.go

```diff
@@ -82,19 +81,27 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 		logp.Err("Could not start registrar: %v", err)
 	}
 	// Stopping registrar will write last state
-	defer registrar.Stop()
+	defer func() {
```

Comment:
This change here should not matter much, due to registrar normally not being blocked for long times. But `registrarLogger` using a channel + if registrar would block, publisher would be blocked as well.